src/interfaces/python/swig_typemaps.i

```diff
@@ -1292,5 +1292,30 @@ TYPEMAP_SPARSEFEATURES_OUT(PyObject,      NPY_OBJECT)
     PyErr_SetString(PyExc_SystemError, $1.what());
     SWIG_fail;
 }
+%rename(_kernel) kernel;
+
+%pythoncode %{
+def _internal_factory_wrapper(object_name, new_name, docstring=None):
+    """
+    A wrapper that returns a generic factory that
+    accepts kwargs and passes them to shogun.object_name
+    via .put
+    """
+    _obj = getattr(_shogun, object_name)
+    def _internal_factory(name, **kwargs):
+
+        new_obj = _obj(name)
+        for k,v in kwargs.items():
```

Comment:
is this python2 and 3 ompatible? I remember something changed in iterating over dictionaries ...

Reply:
ah yeah, dont worry about it